### ISSUE TITLE
Can't access variables if you switch between stackframes several times

### ISSUE THREAD

**Issue by jack-jjm** [elazarcoh/simply-view-image-for-python-debugging#15]
Say we're debugging the following file:

```python
import cv2

def subprocedure():
    image2 = cv2.imread("image.png")
    pass

def procedure():
    image = cv2.imread("image.png")
    subprocedure()

procedure()
pass
```

We run it, and set a breakpoint at line 5, the `pass` in `subprocedure()`.  The "call stack" widget on the left side of the screen will look like this:

![](https://i.imgur.com/Gdd7R9H.png)

Currently, I can view the `image2` variable if I put my caret over it and use the code action that comes up. I *can't* view the `image` variable in the body of `procedure()`, because it's out of scope. If I click on `procedure` in the call stack, I switch to that, and SimplyView keeps up just fine: it switches scope and I can now view `image` (and no longer `image2`).

*But*, if I now switch *back* to `subprocedure`, I still can't view `image2`, I can only view `image`. Basically, SimplyView is stuck in the scope of `procedure`. It can switch stack frames once, but not twice.

I actually am trying to develop my own image viewing extension, and I came up against this same problem. When I looked at this codebase to see how you solved the problem, I realized that you didn't, because if I understand correctly you sniff DAP `variables` requests to detect when the user switches stack frames, but those requests only seem to come through on the first switch, for some reason. Maybe it worked in a previous version of VS code?

In any case, what I really wanted to do was ask you if you ever came up with any ideas for how to solve this problem?

**Comment by elazarcoh** [comment]
Yes, I did encounter this exact issue. In fact, I mentioned it in the Q&A section in the README.

I never solved it. The only real solution that I can think of, is asking ms in the vscode repository to provide an API to get the current stack frame somehow. I tried to search something like this but couldn't find anything, not in the API and not in the debug-protocol.

So, the current workaround is to force vscode request the variables of the current frame. You can see [here](https://github.com/elazarcoh/simply-view-image-for-python-debugging/blob/b2a8f3727e6c180183b6e232cb765a8832dcc0e4/src/extension.ts#L254) how I did it. 
This solution requires manual user action. Maybe I'll do it automatically, I'm not yet sure if that's a good idea.

**Comment by geajack** [comment]
It seems there's been an issue open on it since 2018. https://github.com/microsoft/vscode/issues/63943

**Comment by elazarcoh** [comment]
seems like someone just self-assigned this, so I keep my hopes up :)

**Comment by elazarcoh** [comment]
Just saw the response in the upstream issue. That's unfortunate. Even if I would make PR in the VSCode repo, I doubt it'll get merged. But I'll give it a try. No promises.

**Comment by elazarcoh** [comment]
OK. I implemented this feature at my fork, and it seems to be working (I get the correct frame-id). Now I just need to figure out the correct way to do this, because now I don't think it's a good way to access this info.

**Comment by neriyaco** [comment]
I don't know if this issue is still relevant, but the API is added [here](https://github.com/microsoft/vscode/issues/63943#issuecomment-2098933660)

**Comment by elazarcoh** [comment]
Yes, I have it on my list.
 But since vscode emits a variable request when the frame is changed then it is mostly solved already. But still I wish to add it sometime.

**Comment by elazarcoh** [comment]
Will be added in the next release
